packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.styles.ts

```diff
@@ -135,6 +135,15 @@ export const getStyles = (props: IMessageBarStyleProps): IMessageBarStyles => {
         display: 'flex',
         wordBreak: 'break-word',
         selectors: {
+          // In high contrast mode, Links within MessageBars must revert to the default color
+          // defined by the high contrast settings rather than following the theme colors.
+          '& .ms-Link': {
```

Comment:
Thanks @ecraig12345 for fixing this. 

Wondering what the best practice is in a case like this... I assumed that using a Customizer component with scoped settings might be a little more robust than depending on the class name. Thoughts?

Reply:
Interesting question. We've typically used class names for nested styling, but you're right that the Customizer approach is probably more robust (except in cases where the extra selector specificity is necessary for some reason). Some concerns about if we started to use that approach more widely though:
- Perf impact--may not be much, but it's hard to say for sure without testing. (Though perf impact for this specific case should be minimal since there should only be one MessageBar on a page, or worst case a couple of them.) 
- Not ideal to have some of the files in the .styles.ts and others in the main file
- Not sure whether it would conflict with the effort @xugao may be starting to move styles back to scss

Also realized thinking about it more that this style could potentially be moved to Link itself instead of being an override in MessageBar.  It wouldn't do anything except in cases like MessageBar where a parent of the Link has set MsHighContrastAdjust to none.

Reply:
this made me notice... what is this code here: https://github.com/microsoft/fluentui/blob/master/packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.base.tsx#L155

Let's just keep the fix simple here and not use `Customizer`. it doesn't really make this particular case more robust but it brings more complexity.

Reply:
>  it doesn't really make this particular case more robust

@xugao I'm just passing through here, and don't know what is planned for the future, but there are always trade-offs.  

There has clearly been an effort to create a public interface for style overrides through the styles prop. Bypassing this interface means that when Link is changed in the future there are no guarantees that this type a css override will continue to work.  Or, worse in my mind is that future developers will feel like Link is too fragile to change due to all the unreferenced css overrides.

`Customizer` is the solution that consumers would need to use for this scenario.  Is it really more complex?  Meaning that it would be more difficult for future developers to understand the intentions of the code?  Can you please help me understand?

Reply:
Talked with @dzearing about this and while he agrees that the Customizer approach is more robust than class names, he thinks we should stick with class names for now for a few different reasons, mainly revolving around the upcoming move away from css-in-js (due to perf concerns) to scss and css variables: (this was copied from a chat)
1. we're moving to css variables so this is temporary
2. it will be easier to rationalize when we do
3. easier to upgrade users if we're not very reliant on context for passing around css

(I think what I said above about sticking with the same approach we use elsewhere in our codebase and keeping it centralized in the styles file was also a consideration.)

Reply:
> stick with class names

Maybe I'm trying to solve pain points that don't actually exist for you all, but maybe you'd want to consider importing the string for the class name from somewhere rather than just hardcoding it.

Reply:
Just to chime in and support @ecraig12345 - we definitely want to consider the future here.

We know that css in js leads to performance problems when done in bulk; it's easy to abuse as well. We know that relying on react context to spread styling rules ends up causing react re-renders when themes change. We've worked around these pitfalls in numerous ways over the past years through multiple class caching layers. We used css in js primarily for 2 reasons: normalizing specificity and dynamic contextual theming. But everything has tradeoffs, and we're just not happy with the approach.

CSS custom properties really help with contextual theming. You can define variables at pretty much any level and they can cascade down. We could use this concept to apply standardized classes which represent themable values. Components like Link and Button could consume these tokens and provide fallbacks. This means from a perf perspective:

1. You don't need to re-evaluate the stylesheet. It can be static, but still customizable.
2. You don't depend on react context for theming. No react context means you could even render content using any other css tech (static css included) inside of the controls and still use the theme values.
3. It also means that if the theme changes, you don't need to re-render everything. You just spit out new variable values and they trickle down the dom tree.

So, that's the future. In the present, we still have Customizer, styles props, css in js, etc. If we use the approach of styling subcomponents as @ecraig12345 did here, it means:

1. As she mentioned, having all the styling logic is in one place makes it much easier to port the concern.
2. We achieve the desired outcome (links get styled correctly)
3. Porting to css variables would remove the react context dependency in delivering theme values to the styles.

It certainly has a lot of downsides too. Global selectors are bad. Reaching into internals is bad. We need to move to variables as soon as possible. This is our current focus in `@fluentui/react-next` right now (our v8 alpha build)